dbt/adapters/duckdb/secrets.py

```diff
@@ -0,0 +1,155 @@
+from dataclasses import dataclass
+from dataclasses import fields
+from enum import Enum
+from typing import Optional
+from typing import Union
+
+from dbt_common.dataclass_schema import dbtClassMixin
+
+
+DEFAULT_SECRET_PREFIX = "_dbt_secret_"
+
+
+class SecretType(Enum):
+    S3 = 0
+    AZURE = 1
+    R2 = 2
+    GCS = 3
+    HUGGINGFACE = 4
+
+
+class SecretProvider(Enum):
+    CONFIG = 0
+    CREDENTIAL_CHAIN = 1
+
+
+class AzureSecretProvider(Enum):
+    CONFIG = 0
+    CREDENTIAL_CHAIN = 1
+    SERVICE_PRINCIPAL = 2
+
+
+@dataclass
+class Secret(dbtClassMixin):
+    type: SecretType
+    persistent: bool = False
+    name: Optional[str] = None
+    provider: Optional[Union[SecretProvider, AzureSecretProvider]] = None
+    scope: Optional[str] = None
+
+    @classmethod
+    def cls_from_type(cls, secret_type: Optional[SecretType]):
```

Comment:
Hmm, I see your point. It's probably easier to just let `duckdb` handle the situation where someone enters an unsupported secret type/parameter instead of replicating that logic here.